src/Adapter/Order/CommandHandler/UpdateProductInOrderHandler.php

```diff
@@ -83,6 +93,9 @@ public function handle(UpdateProductInOrderCommand $command)
                 $orderInvoice = new OrderInvoice($command->getOrderInvoiceId());
             }
 
+            // Set Context Shop
+            $this->contextStateManager->setShop(new Shop($orderDetail->id_shop));
```

Comment:
There is something weird here, this handler didn't have the state manager because it is actually handled internally by `OrderProductQuantityUpdater` which already overrides the Shop context based on the info from `$cart->id_shop`

So there are two weird things:
- the shop should be overridden when `$this->orderProductQuantityUpdater->update` is called so it will erase the previous override from this class, therefore it should be useless here
- if there is a bug and the wrong shop was used then does it means that `$orderDetail->id_shop` is different from `$cart->id_shop` ?! this is quite worrying 😅

In the end if your fix works then I guess the override was also necessary in the previous calls before `orderProductQuantityUpdater` is used like in `updateSpecificPrice` that would explain why it fixes it

Reply:
Finally if the override shop is needed earlier then there are chances that other context entities also need it (Country, Currency, ...) I recently added some helpers in this PR https://github.com/PrestaShop/PrestaShop/pull/22246 to help doing this, you can use one of these two methods which are quite convenient:

- `AbstractOrderCommandHandler::setCartContext`
- `AbstractOrderCommandHandler::setOrderContext`

Reply:
Fixed